mc.tests/x86-mov.exp

```diff
@@ -0,0 +1,151 @@
+
+}
+
+set suite(x86) {
+    {} {movb %al, %bl} {EBX := (high:24\[EBX\]).(low:8\[EAX\])}
+    {} {movb %ah, %bh} {EBX := (high:16\[EBX\]).((extract: 15:8\[EAX\]).(low:8\[EBX\]))}
+    {} {movb %al, %bh} {EBX := (high:16\[EBX\]).((low:8\[EAX\]).(low:8\[EBX\]))}
+    {} {movb %ah, %bl} {EBX := (high:24\[EBX\]).(extract: 15:8\[EAX\])}
+    {} {movb %al, (%ebx)} {mem32 := mem32 with \[EBX, el\]:u8 <- low:8\[EAX\]}
+    {} {movb %ah, (%ebx)} {mem32 := mem32 with \[EBX, el\]:u8 <- extract: 15:8\[EAX\]}
+
+    {} {movw %ax, %bx} {EBX := (high:16\[EBX\]).(low:16\[EAX\])}
+    {} {movw %ax, (%ebx)} {mem32 := mem32 with \[EBX, el\]:u16 <- low:16\[EAX\]}
+    {} {movl %eax, %ebx} {EBX := EAX}
+    {} {movl %eax, (%ebx)} {mem32 := mem32 with \[EBX, el\]:u32 <- EAX}
+
+    {0x8a,0xd8} {movb %al, %bl} {EBX := (high:24\[EBX\]).(low:8\[EAX\])}
+    {0x8a,0xfc} {movb %ah, %bh} {EBX := (high:16\[EBX\]).((extract: 15:8\[EAX\]).(low:8\[EBX\]))}
+    {0x8a,0xf8} {movb %al, %bh} {EBX := (high:16\[EBX\]).((low:8\[EAX\]).(low:8\[EBX\]))}
+    {0x8a,0xdc} {movb %ah, %bl} {EBX := (high:24\[EBX\]).(extract: 15:8\[EAX\])}
+    {} {movb (%ebx), %al} {EAX := (high:24\[EAX\]).(mem32\[EBX, el\]:u8)}
+    {} {movb (%ebx), %ah} {EAX := (high:16\[EAX\]).((mem32\[EBX, el\]:u8).(low:8\[EAX\]))}
+
+    {0x66,0x8b,0xd8} {movw %ax, %bx} {EBX := (high:16\[EBX\]).(low:16\[EAX\])}
+    {} {movw (%ebx), %ax} {EAX := (high:16\[EAX\]).(mem32\[EBX, el\]:u16)}
+    {0x8b,0xd8} {movl %eax, %ebx} {EBX := EAX}
+    {} {movl (%ebx), %eax} {EAX := mem32\[EBX, el\]:u32}
+
+    {} {movw %fs, %ax} {EAX := (high:16\[EAX\]).FS}
+    {} {movl %fs, %eax} {EAX := pad:32\[FS\]}
+    {} {movl %fs, (%eax)} {mem32 := mem32 with \[EAX, el\]:u16 <- FS}
+
+    {} {movw %ax, %fs}  {FS := low:16\[EAX\]}
+    {} {movl %eax, %fs} {FS := low:16\[EAX\]}
+    {} {movl (%eax), %fs}  {FS := mem32\[EAX, el\]:u16}
+
+    {} {movb 0x4, %al} {EAX := (high:24\[EAX\]).(mem32\[0x4:32, el\]:u8)}
+
+    {} {movw 0x4, %ax} {EAX := (high:16\[EAX\]).(mem32\[0x4:32, el\]:u16)}
+    {} {movl 0x4, %eax} {EAX := mem32\[0x4:32, el\]:u32}
+    
+    {} {movb %al, 0x4} {mem32 := mem32 with \[0x4:32, el\]:u8 <- low:8\[EAX\]}
+
+    {} {movw %ax, 0x4} {mem32 := mem32 with \[0x4:32, el\]:u16 <- low:16\[EAX\]}
+    {} {movl %eax, 0x4} {mem32 := mem32 with \[0x4:32, el\]:u32 <- EAX}
+
+    {} {movb $0x4, %al} {EAX := (high:24\[EAX\]).0x4:8}
+    {} {movb $0x4, %ah} {EAX := (high:16\[EAX\]).(0x4:8.(low:8\[EAX\]))}
+
+    {} {movw $0x4, %ax} {EAX := (high:16\[EAX\]).0x4:16}
+    {} {movl $0x4, %eax} {EAX := 0x4:32}
+
+    {} {movb $0x4, (%eax)} {mem32 := mem32 with \[EAX, el\]:u8 <- 0x4:8}
+
+    {} {movw $0x4, (%eax)} {mem32 := mem32 with \[EAX, el\]:u16 <- 0x4:16}
+    {} {movl $0x4, (%eax)} {mem32 := mem32 with \[EAX, el\]:u32 <- 0x4:32}
+
+}
+
+
+proc run {arch suite} {
+    set encode "llvm-mc -arch=$arch --show-encoding"
+    set extract  "awk -vFS=\"\[\" -vRS=\"\]\" '\{print \$2\}'"
```

Comment:
well, expect and tcl are supposed to be more powerful, than awk, so using it here is kind of weird. But fine, let's keep it.